src/aws/osml/gdal/sidd_sensor_model_builder.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import Optional, Union
+
+from xsdata.formats.dataclass.parsers import XmlParser
+
+import aws.osml.formats.sidd.models.sidd_v1_0_0 as sidd100
+import aws.osml.formats.sidd.models.sidd_v2_0_0 as sidd200
+import aws.osml.formats.sidd.models.sidd_v3_0_0 as sidd300
+
+from ..photogrammetry import (
+    ChippedImageSensorModel,
+    ImageCoordinate,
+    PlaneProjectionSet,
+    SARImageCoordConverter,
+    SensorModel,
+    SICDSensorModel,
+    WorldCoordinate,
+)
+from .sensor_model_builder import SensorModelBuilder
+from .sicd_sensor_model_builder import poly2d_to_native, xyzpoly_to_native, xyztype_to_ndarray
+
+logger = logging.getLogger(__name__)
+
+
+class SIDDSensorModelBuilder(SensorModelBuilder):
+    """
+    This builder is used to create sensor models for images that have SIDD metadata. The metadata is provided
+    as XML that conforms to the SIDD specifications. We intend to support multiple SIDD versions but the current
+    software was implemented using the v2.0.0 and v3.0.0 specifications.
+
+    Note that the SIDD sensor models rely heavily on the SICD projections so the class of the returned model
+    will be a SICDSensorModel. Future versions may rename this to SISensorModel or SARSensorModel.
+    """
+
+    def __init__(self, sidd_xml: str):
+        """
+        Construct the builder given the SIDD XML.
+
+        :param sidd_xml: the XML string
+        """
+        super().__init__()
+        self.sidd_xml = sidd_xml
+
+    def build(self) -> Optional[SensorModel]:
+        """
+        Attempt to build a precise SAR sensor model. This sensor model handles chipped images natively.
+
+        :return: the sensor model; if available
+        """
+        try:
+            if self.sidd_xml is None or len(self.sidd_xml) == 0:
```

Comment:
I would use `if not self.sidd.xml:` and eliminate the rest according to the [PEPS8](https://peps.python.org/pep-0008/)

```
For sequences, (strings, lists, tuples), use the fact that empty sequences are false:

# Correct:
if not seq:
if seq:
# Wrong:
if len(seq):
if not len(seq):
```